### ISSUE TITLE
How to install dependencies?

### ISSUE THREAD

**Issue by hemingkx** [xwgeng/RewriteNAT#1]
Hi all,

Thanks for your awesome work and codes!

I tried to run the code and used the following commands to build the environment:
```
conda create -n RewirteNAT python=3.6
conda activate RewirteNAT
conda install pytorch==1.2.0 torchvision==0.4.0 cudatoolkit=10.0 -c pytorch
cd RewirteNAT
pip install --editable ./
```

When `pip install --editable ./` ran, I got errors like this:
```
    ERROR: Command errored out with exit status 1:
     command: /home/azureuser/miniconda3/envs/rewrite/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/azureuser/rewrite/setup.py'"'"'; __file__='"'"'/home/azureuser/rewrite/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: /home/azureuser/rewrite/
    Complete output (14 lines):
    running develop
    running egg_info
    creating fairseq.egg-info
    writing fairseq.egg-info/PKG-INFO
    writing dependency_links to fairseq.egg-info/dependency_links.txt
    writing entry points to fairseq.egg-info/entry_points.txt
    writing requirements to fairseq.egg-info/requires.txt
    writing top-level names to fairseq.egg-info/top_level.txt
    writing manifest file 'fairseq.egg-info/SOURCES.txt'
    reading manifest file 'fairseq.egg-info/SOURCES.txt'
    writing manifest file 'fairseq.egg-info/SOURCES.txt'
    running build_ext
    cythoning fairseq/data/data_utils_fast.pyx to fairseq/data/data_utils_fast.cpp
    error: /home/azureuser/rewrite/fairseq/data/data_utils_fast.pyx
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/azureuser/miniconda3/envs/rewrite/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/home/azureuser/rewrite/setup.py'"'"'; __file__='"'"'/home/azureuser/rewrite/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.
```

Could you give me some instructions to build your environment?

Thanks,
hemingkx

**Comment by xwgeng** [comment]
The reason is that we remove unused files such as `data_utils_fast.pyx`, but not delete the corresponding instructions (lines 52-63) in `setup.py`. Just remove the lines 52-63, and the installation will success.

**Comment by hemingkx** [comment]
Thanks! It succeeded! By the way, can you provide more details about the hyperparameters you use or post some training scripts with hyperparameters? I noticed that the scripts you released do not include them.

**Comment by xwgeng** [comment]
The proposed RewriteNAT is trained with a batch size of 128k for 300k updates except for en->fr (500k). The model with `--max-iter`as 4  and `--share-all-embeddings` (except for en->zh) can obtain the expected result even in the default settings of other hyperparameters (e.g.,`--discriminator-loss-factor`, `--discriminator-scale` and `--generator-scale`). Considering the training efficiency,  the proposed architecture can be learned with two training iterations (`--max-iter 2`), yielding about ~27.4 BLEU on en->de.
